### ISSUE TITLE
Workflow graphs

### ISSUE THREAD

**Issue by cfhammill** [tweag/funflow#152]
Presently funflow has support for transforming workflows into graphs via `toDiagram` (although this isn't exported).

It would be nice to be able to turn funflow graphs into DOT files. The translation from `Diagram` to DOT should be simple, I'm happy to do that and submit a PR. 

A small extension that I think would make the generated graphs cleaner is to be able to annotate `mapA`s so that mapped actions can be optionally collapsed to simplify the graph.

This would require extending `Flow'` to have `MapA` constructor. Does that sound like a reasonable way to add this feature?

**Comment by nc6** [comment]
Yes, I think having `MapA` directly as a constructor for the flow would make a lot of sense. At the moment the implementation of `mapA` will cause the program to hang indefinitely if one tries to print it, which is clearly non-optimal. We've had this on the list of things to do for a while (but don't have any active projects using funflow, so it's not being heavily developed rn)

Likewise, DOT support seems great if you felt like implementing it!
